File: Data Structures & Algorithms/find-duplicate-integer/submission-4.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int fast=0,slow=0;
        while(true){
            slow=nums[slow];
            fast= nums[nums[fast]];
            if(slow==fast)
                break;
        }
        int slow2=0;
        while (true){
            slow2=nums[slow2];
            slow=nums[slow];
            if (slow==slow2)
                break;
        }
        return slow2;
    }
};

File: Data Structures & Algorithms/house-robber/submission-4.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n=len(nums)
        dp=nums
        for i in range(1,n):
            if i==1:
                dp[i]=max(dp[i-1],nums[i])
                continue
            dp[i]=max(dp[i-1],dp[i-2]+nums[i])

        return dp[n-1]